curl/tests/request_test.cc

```diff
@@ -0,0 +1,116 @@
+#include <catch.hpp>
+#include "mock_curl.hpp"
+#include <leatherman/curl/client.hpp>
+#include <leatherman/curl/request.hpp>
+
+using namespace std;
+
+namespace leatherman { namespace curl {
+
+    struct mock_client : client {
+        curl_handle const& get_handle() { return client::get_handle(); }
+    };
+
+    TEST_CASE("curl::request") {
+        mock_client test_client;
```

Comment:
You don't need a test_client for the first set of test, I'd break them into separate cases. And if you're testing request, just create the request object before the sections.